### ISSUE TITLE
`used-dummy-variable` (`RUF052`) - false positive on private global variable when it has a usage in two functions

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#14996]
```py
_bar: int


def foo():
    global _bar # error: Local dummy variable `_bar` is accessed
    _bar = 1


def baz():
    print(_bar)
```

https://play.ruff.rs/7b9e5645-c159-4e00-8b84-49183a6c31c4

**Comment by MichaReiser** [comment]
Thanks
